ThreadGroup/Managers/iotivity/TGNetworkCallbackResult.m

```diff
@@ -0,0 +1,140 @@
+//
+
+#import "TGNetworkCallbackResult.h"
+
+@implementation TGNetworkCallbackResult
+
+- (instancetype)initWithArguments:(va_list)args {
+    NSAssert(NO, @"Must instantiate a sublass of this class");
+    return nil;
+}
+
+@end
+
+@implementation TGNetworkCallbackComissionerPetitionResult
+
+- (instancetype)initWithArguments:(va_list)args {
+    self = [super init];
+    if (self) {
+        NSString *commissionerIdentifier = [NSString stringWithUTF8String:va_arg(args, char *)];
+        NSInteger commissionerSessionIdentifier = (NSInteger)va_arg(args, int);
+        BOOL authorizationFailed = (BOOL)va_arg(args, int);
+        
+        [self setCommissionerIdentifer:commissionerIdentifier];
+        [self setCommissionerSessionIdentifier:commissionerSessionIdentifier];
+        [self setHasAuthorizationFailed:authorizationFailed];
+    }
+    return self;
+}
+
+@end
+
+@implementation TGNetworkCallbackJoinerFinishedResult
+
+- (instancetype)initWithArguments:(va_list)args {
+    self = [super init];
+    if (self) {
+        NSString *joinerIdentifier = [NSString stringWithUTF8String:va_arg(args, char *)];
+        MCState_t state = (MCState_t)va_arg(args, int);
+        NSString *provisioningURL = [NSString stringWithUTF8String:va_arg(args, char *)];
+        NSString *vendorName = [NSString stringWithUTF8String:va_arg(args, char *)];
+        NSString *vendorModel = [NSString stringWithUTF8String:va_arg(args, char *)];
+        NSString *vendorSoftwareVersion = [NSString stringWithUTF8String:va_arg(args, char *)];
+        
+        [self setJoinerIdentifier:joinerIdentifier];
+        [self setState:state];
+        [self setProvisioningURL:provisioningURL];
+        [self setVendorName:vendorName];
+        [self setVendorModel:vendorModel];
+        [self setVendorSoftwareVersion:vendorSoftwareVersion];
+    }
+    return self;
+}
+
+@end
+
+@implementation TGNetworkCallbackJoinResult
+
+- (instancetype)initWithArguments:(va_list)args {
+    self = [super init];
+    if (self) {
+        NSString *provisioningURL = [NSString stringWithUTF8String:va_arg(args, char *)];
+        [self setProvisioningURL:provisioningURL];
+    }
+    return self;
+}
+
+@end
+
+@implementation TGNetworkCallbackErrorResult
+
+- (instancetype)initWithArguments:(va_list)args {
+    self = [super init];
+    if (self) {
+        NSInteger mcResult = (NSInteger)va_arg(args, int);
+        NSInteger caResponseResult = (NSInteger)va_arg(args, int);
+        NSString *token = [NSString stringWithUTF8String:va_arg(args, char *)];
+        self.mcResult = mcResult;
+        self.caResult = caResponseResult;
+        self.token = token;
+    }
+    return self;
+}
+
+@end
+
+@implementation TGNetworkCallbackFetchSettingResult
+
+- (instancetype)initWithArguments:(va_list)args {
+    self = [super init];
+    if (self) {
+        self.parameterIdentifier = (MCMgmtParamID_t)va_arg(args, int);
+        id returnValue;
```

Comment:
I'm a bit confused on the use of returnValue. In some cases, you set returnValue to be something (ie. returnValue = @(val)), but nothing is done with it. Do you use it somewhere else that I'm not seeing?